dashboard/serializers_test.py

```diff
@@ -44,47 +71,107 @@ def _generate_cached_enrollments(user, program, num_course_runs=1, data=None):
+                )
+            )
+            cls.current_grades.append(
+                CachedCurrentGradeFactory.create(
+                    user=cls.user,
                     course_run=enrollment.course_run,
-                    data={'grade': certificate_grades[i]}
+                    data={
+                        "passed": True,
+                        "percent": cls.current_grades_vals[i],
+                        "course_key": enrollment.course_run.edx_course_key,
+                    }
                 )
             )
-        cls.program_enrollment = ProgramEnrollment.objects.create(user=profile.user, program=program)
+        cls.program_enrollment = ProgramEnrollment.objects.create(user=cls.user, program=program)
 
     def test_full_program_user_serialization(self):
         """
         Tests that full ProgramEnrollment serialization works as expected
         """
-        user = self.program_enrollment.user
         program = self.program_enrollment.program
         assert UserProgramSearchSerializer.serialize(self.program_enrollment) == {
             'id': program.id,
-            'enrollments': list(CachedEnrollment.active_data(user, program)),
-            'certificates': list(CachedCertificate.active_data(user, program)),
+            'enrollments': list(CachedEnrollment.active_data(self.user, program)),
+            'certificates': list(CachedCertificate.active_data(self.user, program)),
+            'current_grades': list(CachedCurrentGrade.active_data(self.user, program)),
             'grade_average': 75,
             'is_learner': True
         }
 
-        # when user has staff role, assert that he is not learner.
+    def test_full_program_user_serialization_financial_aid(self):
+        """
+        Tests that full ProgramEnrollment serialization works as expected
+        for financial aid programs.
+        the difference with test_full_program_user_serialization
+        is that the grade is calculated using the current grades
+        """
+        program = create_program()
+        create_tier_programs(program)
+        program_enrollment = ProgramEnrollment.objects.create(user=self.user, program=program)
+        enrollments = self._generate_cached_enrollments(self.user, program, num_course_runs=2)
+        current_grades = []
+        for i, enrollment in enumerate(enrollments):
+            order = OrderFactory.create(user=self.user, status='fulfilled')
+            LineFactory.create(order=order, course_key=enrollment.course_run.edx_course_key)
+            current_grades.append(
+                CachedCurrentGradeFactory.create(
+                    user=self.user,
+                    course_run=enrollment.course_run,
+                    data={
+                        "passed": True,
+                        "percent": self.current_grades_vals[i],
+                        "course_key": enrollment.course_run.edx_course_key,
+                    }
+                )
+            )
```

Comment:
you have `setUpTestData` creating data only for `test_full_program_user_serialization`, then for this method you perform similar setup steps in the body of the test case function. a cleaner approach might be to create everything up to the `Program`s in the `setUpTestData` method (and anything else that all test cases need), then have separate helper methods for creating the enrollments/grades/etc for the separate test cases. i suggest creating the `Program`s in  `setUpTestData` b/c it will ensure that they will be deleted after every test case, and that will cascade down to delete all the associated edx data